sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/SelectQuery.cs

```diff
@@ -0,0 +1,135 @@
+            return _innerQuery;
+        }
+
+        /// <summary>
+        /// Used when applying the TOP() aggregate from the ADT query language. Same functionality as select
+        /// but inserts TOP() into the query structure as well.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), ie the number of instances to return. </param>
+        /// <param name="args"> The arguments that can be optionally passed with top (eg property name). </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectTop(int count, params string[] args)
+        {
+            // turn into correct format -- eg. SELECT TOP(3)
+            StringBuilder topArg = new StringBuilder().Append("TOP").Append('(').Append(count).Append(')');
+
+            // account for optional arguments
+            if (args.Length != 0)
+            {
+                // TODO -- topargs.append.string.format.... (way around loop)
+                // append first argument manually since no comma needed
+                topArg.Append(' ').Append(args[0]);
+
+                // append other arguments separated by commas
+                for (int i = 1; i < args.Length; i++)
+                {
+                    topArg.Append(", ").Append(args[i]);
+                }
+            }
+
+            _clause = new SelectClause(new string[] { topArg.ToString() });
+            return _innerQuery;
+        }
+
+        /// <summary>
+        /// Used when applying the COUNT() aggregate from the ADT query language. Similar to SelectTop().
+        /// </summary>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectCount()
+        {
+            string countArg = new StringBuilder().Append("COUNT").Append('(').Append(')').ToString();
+
+            _clause = new SelectClause(new string[] { countArg });
+            return _innerQuery;
+        }
+
+        /// <summary>
+        /// Called when overriding the query builder with the literal query string.
+        /// </summary>
+        /// <param name="literalQuery"> Query in string format. </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectOverride(string literalQuery)
+        {
+            // TODO -- just use select?
+            // approach 1 - method overload
+            // approach 2 - seperate method
+            _clause = new SelectClause(new string[] { literalQuery });
+
+            return _innerQuery;
+        }
+
+        /// <inheritdoc/>
+        public override AdtQueryBuilder Build()
+        {
+            throw new InvalidOperationException("You can't just have a select statement");
+        }
+
+        /// <inheritdoc/>
+        public override string Stringify()
+        {
+            StringBuilder selectComponents = new StringBuilder();
+            selectComponents.Append("SELECT ");
```

Comment:
I don't feel super strongly about this, but I recommend you create a static class with string constants that hold on to query key words.

Reply:
and reference them instead of typing the keywords everytime. This will help with refactoring and consistency.